internal/engine/buildcontroller_test.go

```diff
@@ -245,6 +246,51 @@ func TestBuildControllerCrashRebuild(t *testing.T) {
 	f.assertAllBuildsConsumed()
 }
 
+func TestCrashRebuildTwoContainersOneImage(t *testing.T) {
+	f := newTestFixture(t)
+	defer f.TearDown()
+
+	manifest := manifestbuilder.New(f, "sancho").
+		WithK8sYAML(testyaml.SanchoTwoContainersOneImageYAML).
+		WithImageTarget(NewSanchoLiveUpdateImageTarget(f)).
+		Build()
+	f.Start([]model.Manifest{manifest}, true)
+
+	call := f.nextCall()
+	assert.Equal(t, manifest.ImageTargetAt(0), call.image())
+	f.waitForCompletedBuildCount(1)
+
+	f.b.nextLiveUpdateContainerIDs = []container.ID{"c1", "c2"}
+	f.podEvent(podbuilder.New(t, manifest).
+		WithContainerID("c1").
+		WithContainerIDAtIndex("c2", 1).
+		Build())
+	f.fsWatcher.events <- watch.NewFileEvent(f.JoinPath("main.go"))
+
+	call = f.nextCall()
+	f.waitForCompletedBuildCount(2)
+	f.withManifestState("sancho", func(ms store.ManifestState) {
+		assert.Equal(t, 2, len(ms.LiveUpdatedContainerIDs))
+	})
+
+	// Restart one container id but not the other.
```

Comment:
i don't think you mean "restart"...? maybe:

> // Simulate pod event where one of the containers has been replaced

?

Reply:
AFAICT, "restart" is consistent with the terminology that Kubernetes uses

if you kill the main container process, Kubernetes creates a new container ID and increments the restartCount field to 1

Reply:
Ah. That makes sense -- but fwiw I read "restart" like `restart_container`, i.e. it's the same container, we just re-exec the entrypoint. So i might consider replacing cuz that word has a different meaning in our codebase. but nbd